### ISSUE TITLE
Flickering gallery

### ISSUE THREAD

**Issue by Voyz** [miromannino/Justified-Gallery#142]
Using JG with AJAX pagination for images I encountered strange flicking. After a day in your code I realised it is caused by the flushRow() and rewind() functions, and by the refreshSensivity.

In flushRow() we set the height of the gallery:
```
this.$gallery.height(this.offY + this.buildingRow.height + 
    this.border + (this.isSpinnerActive() ? this.getSpinnerHeight() : 0));
```

The ```this.offY``` is modified in rewind():
```
this.offY = this.border;
```
Where ```this.border``` is normally small. This pretty much means whole gallery is collapsing to a narrow horizontal bar.

.

In case the web page is short enough and the screen is small enough the fully build gallery will occupy more than the window's height and the browser will display the scrollbar. 

However when we rewind the height is reset to that narrow bar which causes the scrollbar to disappear for an instant, until the gallery will be rebuilt.

Now, that scrollbar effectively causes the width of your body to be some 10-20px smaller. However, using the checkWidth() the gallery will detect any width changes unless you set refreshSensitivity to a number like 30, which by default is 0.

So summing up, whenever the gallery refreshes it shrinks down to the small bar, then expands to its full width, resulting in the scrollbar disappearing and appearing again. That causes the gallery's width to change which is caught by checkWidth(). That in turn triggers gallery's rewind and refresh and we're back at the start. And that causes flicker.

I'm attaching a modified version that prevents this behaviour in two ways:
[jquery.justifiedGallery.js.zip](https://github.com/miromannino/Justified-Gallery/files/89212/jquery.justifiedGallery.js.zip)

 1. I don't resize the gallery in flushRow(), but simply store its new size and set it at the end of the process.
 2. I check if that scrollbar appears and prevent checkWidth() from rewinding and refreshing.

Cheers
Voy

Code changed:
```
...
  function hasScrollBar() {
    return $("body").height() > $(window).height();
  }

...

//Gallery Height
    // this.$gallery.height(this.offY + this.buildingRow.height + 
        // this.border + (this.isSpinnerActive() ? this.getSpinnerHeight() : 0));
    this.galleryHeightToSet = this.offY + this.buildingRow.height + 
        this.border + (this.isSpinnerActive() ? this.getSpinnerHeight() : 0);

...

 /**
   * Checks the width of the gallery container, to know if a new justification is needed
   */
  var scrollBarOn = false;
  JustifiedGallery.prototype.checkWidth = function () {
    this.checkWidthIntervalId = setInterval($.proxy(function () {
      var galleryWidth = parseFloat(this.$gallery.width());
      if (hasScrollBar() === scrollBarOn){

        if (Math.abs(galleryWidth - this.galleryWidth) > this.settings.refreshSensitivity) {
          this.galleryWidth = galleryWidth;
          this.rewind();

          // Restart to analyze
          this.startImgAnalyzer(true);
        }
      } else {
        scrollBarOn = hasScrollBar();
        this.galleryWidth = galleryWidth;
      }
    }, this), this.settings.refreshTime);

  };

...

    /* Stop, if there is, the timeout to start the analyzeImages.
     This is because an image can be set loaded, and the timeout can be set,
     but this image can be analyzed yet.
     */
    this.stopImgAnalyzerStarter();

    //On complete callback
    this.$gallery.trigger(isForResize ? 'jg.resize' : 'jg.complete');
    this.$gallery.height(this.galleryHeightToSet);
  };

...
```

**Comment by phelma** [comment]
@Voyz Thanks for this, I have created a PR with your fix:  #151

**Comment by Voyz** [comment]
@phelma Glad I could help. Sorry I couldn't make a pull request nor write tests; Had to put the code together quickly for a fix. Happy you found it useful.
